membership-attribute-service/app/prodtest/InstanceCountOnSchedule.scala

```diff
@@ -0,0 +1,44 @@
+package prodtest
+
+import akka.actor.ActorSystem
+import com.gu.memsub.util.ScheduledTask
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.DurationInt
+import com.amazonaws.services.autoscaling.AmazonAutoScalingClientBuilder
+import com.gu.aws.CredentialsProvider
+import com.typesafe.scalalogging.LazyLogging
+import scala.collection.JavaConverters._
+
+class InstanceCountOnSchedule(stage: String)(implicit ec: ExecutionContext, system: ActorSystem) extends LazyLogging {
+  private val defaultInstanceCount = 6
+
+  private lazy val autoScalingClient =
+    AmazonAutoScalingClientBuilder
+      .standard()
+      .withRegion("eu-west-1")
+      .withCredentials(CredentialsProvider)
+      .build()
+
+  private def getCurrentNumberOfInstances(stage: String): Int =
+    autoScalingClient
+      .describeAutoScalingGroups()
+      .getAutoScalingGroups
+      .asScala
+      .toList
+      .filter(_.getAutoScalingGroupName.startsWith(s"Memb-Attributes-$stage"))
+      .map(_.getInstances.asScala.size)
```

Comment:
does the `getInstances` need to be filtered on those which are _active/healthy_ (to prevent it spiking during a release - since the number of instances will double for a while)

Reply:
if that wouldn't help - perhaps instead we should be asking the load balancer how many instances it is sending traffic to?